controllers/argocd/argocd_controller.go

```diff
@@ -104,90 +106,109 @@ func (r *ReconcileArgoCD) Reconcile(ctx context.Context, request ctrl.Request) (
 		return reconcile.Result{}, err
 	}
 
-	newPhase := argocd.Status.Phase
-	// If we discover a new Argo CD instance in a previously un-seen namespace
-	// we add it to the map and increment active instance count by phase
-	// as well as total active instance count
-	if _, ok := ActiveInstanceMap[request.Namespace]; !ok {
-		if newPhase != "" {
-			ActiveInstanceMap[request.Namespace] = newPhase
-			ActiveInstancesByPhase.WithLabelValues(newPhase).Inc()
-			ActiveInstancesTotal.Inc()
-		}
-	} else {
-		// If we discover an existing instance's phase has changed since we last saw it
-		// increment instance count with new phase and decrement instance count with old phase
-		// update the phase in corresponding map entry
-		// total instance count remains the same
-		if oldPhase := ActiveInstanceMap[argocd.Namespace]; oldPhase != newPhase {
-			ActiveInstanceMap[argocd.Namespace] = newPhase
-			ActiveInstancesByPhase.WithLabelValues(newPhase).Inc()
-			ActiveInstancesByPhase.WithLabelValues(oldPhase).Dec()
+	// Fetch the value of the $LABEL_SELECTOR environment variable from the argocd instance
+	var labelSelectorValue string
+	for _, envVar := range argocd.Spec.Server.Env {
+		if envVar.Name == "LABEL_SELECTOR" {
+			labelSelectorValue = envVar.Value
+			break
 		}
 	}
 
-	ActiveInstanceReconciliationCount.WithLabelValues(argocd.Namespace).Inc()
+	// Check if the labelSelectorValue matches the r.LabelSelector (command-line option)
+	if r.LabelSelector != labelSelectorValue {
+		fmt.Println("Skipping reconciliation as labelSelector does not match.")
+		return reconcile.Result{}, fmt.Errorf("Skipping reconciliation as labelSelector does not match for the give ArgoCD instance : %v", argocd.Name)
+	}
 
-	if argocd.GetDeletionTimestamp() != nil {
+	// Check if the $LABEL_SELECTOR environment variable is set in the argocd instance and begin reconcilliation for the same.
+	if r.LabelSelector == labelSelectorValue || labelSelectorValue == common.ArgoCDDefaultLabelSelector {
```

Comment:
This check is not required as the above `if` block will satisfy the required check.